contracts/rewards/src/msg.rs

```diff
@@ -60,4 +61,16 @@ pub enum ExecuteMsg {
 
 #[cw_serde]
 #[derive(QueryResponses)]
-pub enum QueryMsg {}
+pub enum QueryMsg {
+    /// Gets the rewards pool details for the given `pool_id``
+    #[returns(RewardsPool)]
+    RewardsPool { pool_id: PoolId },
+}
+
+#[cw_serde]
+pub struct RewardsPool {
+    pub balance: Uint128,
+    pub epoch_duration: Uint64,
+    pub rewards_per_epoch: Uint128,
+    pub last_distribution_epoch: Option<Uint64>,
```

Comment:
I think we need the current epoch in here too, so that way a caller knows whether rewards need to be distributed. Without knowing the current epoch, I don't think the last distributed epoch is very useful.